### PR TITLE
First version of the cartodbfy requirements

### PR DESCRIPTION
**Pull request by rafatower**
@javisantana @pramsey  can you please review?

This is probably not complete and I know we don't meet some requirements yet, in particular "Cartodbfy shall deal with text columns for imports, regarding CartoDB columns" (which I think is important for importing CSV files without type guessing).

cc/ @iriberri

### CODE REVIEW

**Comment by javisantana** [comment]
FYI, there is a previous version, check the wiki: https://github.com/CartoDB/cartodb-postgresql/wiki/CartoDB-user-table

**Comment by iriberri** [comment]
:clap: :clap:

File: doc/cartodbfy-requirements.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+- Exporting and re-importing the same table in CartoDB shall produce equivalent tables, with the same features associated to the same ``cartodb_id``'s.
+
+Note that there should be only one feature per row in the source table. If there's more than one, then which one is used for ``the_geom`` and ``the_geom_webmercator`` fields is not determined.
+
+
+
+Low-level requirements
+======================
+
+- If the original table contains a valid ``cartodb_id`` column, it shall be used
```
Comment (pramsey) [inline doc/cartodbfy-requirements.rst]: "valid" is a magic word. more detail: integer, unique

File: doc/cartodbfy-requirements.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+- If the original table contains a valid ``the_geom`` or ``the_geom_webmercator`` it shall be used.
```
Comment (pramsey) [inline doc/cartodbfy-requirements.rst]: in this case, "valid" = in the expected projection

File: doc/cartodbfy-requirements.rst [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+The cartodbfycation is the process of converting an arbitrary postgres table into a valid CartoDB table, and register it in the system so that it can be used in the CartoDB editor and platform to generate maps and analysis.
+
+
+
+Valid CartoDB tables
+====================
+
+A valid CartoDB table shall meet the following conditions:
+
+- Have a ``cartodb_id`` column as primary key with a sequence
```
Comment (pramsey) [inline doc/cartodbfy-requirements.rst]: integer requirement, sequence as default value

File: doc/cartodbfy-requirements.rst (hunk 1 above)
Comment (rafatower) [inline doc/cartodbfy-requirements.rst]: right
but text is also valid as long as unique, not null. currently not checking for non-null

**Comment by iriberri** [comment]
I have applied Paul's and Rafa's feedback here. Merging @rafatower?

**Comment by rafatower** [comment]
I'd add a deprecation note in https://github.com/CartoDB/cartodb-postgresql/wiki/CartoDB-user-table

other than that, ok. We can improve from here.

thanks for taking care of this long, overdue task.

**Comment by iriberri** [comment]
Not null issue edited, wiki deprecated.

**Comment by iriberri** [comment]
@rafatower is this okay now?

**Comment by rafatower** [comment]
Yep, I think so.

BTW if you look into the "deprecated" doc, you'll find concerns about the multi-geometry thing we've talked about. I have my views on that (prefer not to comment here).

**Pull request merged**